lisa/tools/modprobe.py

```diff
@@ -63,18 +63,32 @@ def remove(self, mod_names: List[str], ignore_error: bool = False) -> None:
     def load(
         self,
         modules: Union[str, List[str]],
-    ) -> None:
+        dry_run: bool = False,
+    ) -> bool:
+
         if isinstance(modules, list):
             modules_str = "-a " + " ".join(modules)
         else:
             modules_str = modules
-        self.run(
-            f"{modules_str}",
+        command = f"{modules_str}"
+        if dry_run:
+            command = "--dry-run -q " + command
+        result = self.run(
+            command,
             force_run=True,
             sudo=True,
+        )
+        if dry_run:
+            return result.exit_code == 0
+
+        result.assert_exit_code(
             expected_exit_code=0,
-            expected_exit_code_failure_message=f"Fail to load module[s]: {modules_str}",
+            message=f"Fail to load module[s]: {modules_str}",
```

Comment:
Is it worth to output the detail failure message? You can add an argument on `assert_exit_code` to specify if print raw output like `with_output`.